test/Dialect/Arc/arc-canonicalizer.mlir

```diff
@@ -295,3 +295,52 @@ hw.module @DontSinkDifferentConstants1(in %x: i4, out out0: i4, out out1: i4, ou
   hw.output %0, %1, %2 : i4, i4, i4
 }
 // CHECK-NEXT: }
+
+//===----------------------------------------------------------------------===//
+// MergeVectorizeOps
+//===----------------------------------------------------------------------===//
+
+hw.module @VecOpTest(in %b: i8, in %e: i8, in %h: i8, in %k: i8, in %c: i8, in %f: i8,
```

Comment:
Maybe add a test where multiple input vectors of a single `arc.vectorize` come from other vectorize ops and can be merged?

Reply:
Also, it'd be nice to have a test for the case where the results of the vectorize op to be merged into the current one has other users.

Reply:
And also one where the vector is not taken 1-1 but some reshuffling would need to happen.

Reply:
> And also one where the vector is not taken 1-1 but some reshuffling would need to happen.

```
auto otherVecOp = inputVec[0].getDefiningOp<VectorizeOp>();
    auto resultVec = otherVecOp.getResults();
    auto comparator = [](const mlir::Value &a, const mlir::Value &b) {
      return a.getAsOpaquePointer() < b.getAsOpaquePointer();
    };
    std::sort(inputVec.begin(), inputVec.end(), comparator);
    std::sort(resultVec.begin(), resultVec.end(), comparator);
```

Can we sort both? if not do you have any other way?

Reply:
> And also one where the vector is not taken 1-1 but some reshuffling would need to happen.

Is there an easy way to sort `inputVec` and `resVec = otherVecOp.getResults()`?
Maybe I can store inputVec in `DensMap` and check but that will need to specialize the operators.

Reply:
I don't understand why you would need to sort something here. For this test, your patten should not apply, i.e., it should not merge the two `arc.vectorize` ops.